build-system/tasks/helpers.js

```diff
@@ -457,6 +457,7 @@ async function compileUnminifiedJs(srcDir, srcFilename, destDir, options) {
         entryPoints: [entryPoint],
         bundle: true,
         sourcemap: true,
+        format: argv.esm ? 'esm' : 'cjs',
```

Comment:
note: when I had naively tried `esm` earlier I had seen runtime errors.

Reply:
`performBundle()` is only called by `compileUnminifiedJs()` during `gulp build`, where `argv.esm` is disregarded.

For `gulp dist [--esm]`, we'd have to make the change so it's visible to this code:

https://github.com/ampproject/amphtml/blob/4ebe517fe6dbb2038298bbecae3e715576610be0/build-system/compile/compile.js#L343-L373

Reply:
We would need to convert all example pages to use `type=module` for that to work. The point here isn't to actually output ESM (and you can't at the moment, because `--esm` isn't a valid option to `gulp build`), but to move it out of Babel.

The npm scripts I'm building will hook into this to allow an ESM output.